forms-flow-web/src/services/UserService.js

```diff
@@ -61,19 +60,29 @@ const initKeycloak = (store, ...rest) => {
         store.dispatch(setLanguage(KeycloakData.tokenParsed.locale || "en"));
         //Set Cammunda/Formio Base URL
         setApiBaseUrlToLocalStorage();
-
-        let roles = [];
-        for (let i = 0; i < UserRoles.length; i++) {
-          const roleData = ROLES.find((x) => x.title === UserRoles[i]);
-          if (roleData) {
-            roles = roles.concat(roleData.id);
+        
+        store.dispatch(getFormioRoleIds((err,data)=>{
+          if(err){
+            console.error(err);
+          }else{
+             // filter the role based on keyclok role
+             let roles = [];
+               data.forEach((i)=>{
```

Comment:
Try to use meaning full variable names for better understanding. Instead of data say roleData and instead of "i" a relevant name